src/components/side_panel/chart/waterfall_chart/waterfall_chart_design_panel.ts

```diff
@@ -1,22 +1,47 @@
-import { useExternalListener, useState } from "@odoo/owl";
+import { Component, useExternalListener, useState } from "@odoo/owl";
 import {
   CHART_WATERFALL_NEGATIVE_COLOR,
   CHART_WATERFALL_POSITIVE_COLOR,
   CHART_WATERFALL_SUBTOTAL_COLOR,
 } from "../../../../constants";
-import { Color } from "../../../../types";
+import { Color, DispatchResult, SpreadsheetChildEnv, UID } from "../../../../types";
 import { WaterfallChartDefinition } from "../../../../types/chart/waterfall_chart";
-import { GenericChartDesignPanel } from "../line_bar_pie_panel/design_panel";
+import { SidePanelCollapsible } from "../../components/collapsible/side_panel_collapsible";
+import { RoundColorPicker } from "../../components/round_color_picker/round_color_picker";
+import { Section } from "../../components/section/section";
+import {
+  AxisDefinition,
+  AxisDesignEditor,
+} from "../building_blocks/axis_design/axis_design_editor";
+import { GeneralDesignEditor } from "../building_blocks/general_design/general_design_editor";
 import { Checkbox } from "./../../components/checkbox/checkbox";
-import { RoundColorPicker } from "./../../components/round_color_picker/round_color_picker";
 
-export class WaterfallChartDesignPanel extends GenericChartDesignPanel {
+interface Props {
+  figureId: UID;
+  definition: WaterfallChartDefinition;
+  canUpdateChart: (figureID: UID, definition: Partial<WaterfallChartDefinition>) => DispatchResult;
+  updateChart: (figureId: UID, definition: Partial<WaterfallChartDefinition>) => DispatchResult;
+}
+
+export class WaterfallChartDesignPanel extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet-WaterfallChartDesignPanel";
-  static components = { ...GenericChartDesignPanel.components, Checkbox, RoundColorPicker };
+  static components = {
+    GeneralDesignEditor,
+    Checkbox,
+    SidePanelCollapsible,
+    Section,
+    RoundColorPicker,
+    AxisDesignEditor,
+  };
+  static props = {
+    figureId: String,
+    definition: Object,
+    updateChart: Function,
+    canUpdateChart: { type: Function, optional: true },
+  };
 
-  state = useState({ pickerOpened: false });
+  state = useState({ pickerOpened: false, index: 0, currentAxis: "x", activeTool: "" });
```

Comment:
Looks like the state is only used in `closePicker` and `togglePicker` which are themselves unused
```suggestion
```